### PR TITLE
Added Depreciation Percentage as an option

### PR DESCRIPTION
**Pull request by Godmartinz**
# Description
This adds the ability to calculate depreciations by percentage. which takes the purchase cost and grabs the percentage and substitutes it in for the amount. the current formula has not changed. this just adds a private method `calculateDepreciation()` to handle this math.

A percentage option has been added to Depreciations:
image

Floor Value as an amount or percentage:
image

Error if you go outside of 0-100%:
image

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:
* PHP version:
* MySQL version
* Webserver version
* OS version

# Checklist:

- [ ] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [ ] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by snipe — CHANGES_REQUESTED** [review]
This looks great - can you add in a test or two here as well?

**Review by akemidx — APPROVED** [review]
This looks good! Math seems solid and the now() variable seems to be fine for what we wanna do.

**Comment by Godmartinz** [comment]
@snipe added a test for amount and percentage 👍

File: app/Http/Controllers/DepreciationsController.php [hunk 1]
```diff
@@ -62,6 +62,20 @@ public function store(Request $request) : RedirectResponse
...
         $depreciation->months = $request->input('months');
         $depreciation->user_id = Auth::id();
+
+        $request->validate([
+            'depreciation_min' => [
+                'required',
+                'numeric',
+                function ($attribute, $value, $fail) use ($request) {
+                    if ($request->input('depreciation_type') == 'percent' && ($value < 0 || $value > 100)) {
+                        $fail('The depreciation minimum must be between 0 and 100 when depreciation type is percentage.');
```
Comment (snipe) [inline app/Http/Controllers/DepreciationsController.php]: This needs to be localized into english.

File: app/Http/Controllers/DepreciationsController.php [hunk 2]
```diff
@@ -116,6 +130,20 @@ public function update(Request $request, $depreciationId = null) : RedirectRespo
...
         $depreciation->name             = $request->input('name');
         $depreciation->months           = $request->input('months');
+
+        $request->validate([
+            'depreciation_min' => [
+                'required',
+                'numeric',
+                function ($attribute, $value, $fail) use ($request) {
+                    if ($request->input('depreciation_type') == 'percent' && ($value < 0 || $value > 100)) {
+                        $fail('The depreciation minimum must be between 0 and 100 when depreciation type is percentage.');
```
Comment (snipe) [inline app/Http/Controllers/DepreciationsController.php]: Same here

Comment (Godmartinz) [reply app/Http/Controllers/DepreciationsController.php]: it is so 👍

**Pull request merged**